### PR TITLE
 Add two new issue codes and introduce NoDataValidationIssue

### PR DESCRIPTION
**Pull request by joelverhagen**

### CODE REVIEW

File: src/NuGet.Services.Validation.Issues/NoDataValidationIssue.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Services.Validation.Issues
+{
+    public class NoDataValidationIssue : ValidationIssue
+    {
+        public NoDataValidationIssue(ValidationIssueCode issueCode)
```
Comment (loic-sharma) [inline src/NuGet.Services.Validation.Issues/NoDataValidationIssue.cs]: It's not obvious why this type is necessary without looking at `ValidationIssue`'s `Deserialize`. Please leave a comment

File: src/NuGet.Services.Validation.Issues/NoDataValidationIssue.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Services.Validation.Issues
+{
+    public class NoDataValidationIssue : ValidationIssue
```
Comment (loic-sharma) [inline src/NuGet.Services.Validation.Issues/NoDataValidationIssue.cs]: Super nit: we can make this `sealed`

Comment (joelverhagen) [reply src/NuGet.Services.Validation.Issues/NoDataValidationIssue.cs]: I never seal things personally, but sure.

File: src/NuGet.Services.Validation.Issues/NoDataValidationIssue.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Services.Validation.Issues/NoDataValidationIssue.cs]: Sure

**Comment by joelverhagen** [comment]
Progress on https://github.com/NuGet/Engineering/issues/785
Progress on https://github.com/NuGet/Engineering/issues/1047

**Pull request merged**